push-to-s3-and-invalidate-cloudfront/action.yml

```diff
@@ -0,0 +1,148 @@
+    required:  true
+  s3-bucket:
+    description: The name of the S3 bucket
+    required: true
+
+  s3-action:
+    description: |
+      The action to take on the S3 bucket. Typically this should be `update` to push the latest files to S3 or `delete` to delete the folder from S3.
+      If this is not specified, the action will be infered from the $github.event.action variable
+    required: false
+  s3-dir-to-upload-to:
+    description: |
+      The S3 directory to upload the files to. If unspecified, the default `PR-$pull-request-number` will be used
+      or the directory may be infered from the github context. A value of `/` indicates the root of the bucket.
+    required: false
+    default: "pr-previews"
+  s3-dir-to-delete-from:
+    description: |
+      The directory on S3 whose contents need to be deleted. If unspecified, the default `PR-$pull-request-number` will be used
+      or the directory may be infered from the github context
+    required: false
+    default: "pr-previews"
+  s3-files-to-exclude:
+    description: |
+      A pattern or a comma separated list of files to exclude when running the aws s3 sync command. 
+      Refer to https://docs.aws.amazon.com/cli/latest/reference/s3/index.html#use-of-exclude-and-include-filters or more information
+      Set this variable to an empty string if you don't want to exlcude any files
+      If unspecified will default to `pr-previews/*`
+    required: false
+    default: "pr-previews/*"
+  invalidate-cloudfront-cache:
+    description: A flag to indicate if the cloudfront cache should be invalidated
+    required: false
+    default: "false"
+
+outputs:
+  dir_to_upload_to:
+    description: The S3 directory to upload the files to. Computed based on user input
+    value: ${{ steps.upload.dir_to_upload_to }}
+  dir_to_delete_from:
+    description: The S3 directory to upload the files to. Computed based on user input
+    value: ${{ steps.delete.dir_to_delete_from }}
+  files_to_exlude:
+    description: Files to exclude from S3 when using the sync command.
+    value: ${{ steps.exclude.files_to_exclude }}
+
+runs:
+  using: composite
+  steps:
+    - name: Compute the S3 directory to upload to
+      shell: bash
+      id: upload
+      run: |
+        if ${{ inputs.s3-dir-to-upload-to }} == "pr-previews"
+        then
+          dir_to_upload_to=${{ inputs.s3-dir-to-upload-to }}/PR-${{ inputs.pull-request-number }}
+        else if ${{ inputs.s3-dir-to-upload-to }} == "/"
+          dir_to_upload_to=""
+        else
+          dir_to_upload_to=${{ inputs.s3-dir-to-upload-to }}
+        fi
+        echo "dir_to_upload_to=$dir_to_upload_to" >> $GITHUB_OUTPUT
+
+    - name: Compute the S3 directory to delete from
+      shell: bash
+      id: delete
+      run: |
+        if ${{ inputs.s3-dir-to-delete-from }} == "pr-previews"
+        then
+          dir_to_delete_from=${{ inputs.s3-dir-to-delete-from }}/PR-${{ inputs.pull-request-number }}

```

Comment:
Same sugestion as from upload step

Reply:
Same, you can set the `s3-dir-to-delete-from` input variable